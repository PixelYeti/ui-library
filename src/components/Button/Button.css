.button {
	box-sizing: border-box;
	padding: 10px;
	min-width: 100px;

	border-radius: 5px;
	border: none;

	text-align: center;

	font-family: sans-serif;

	color: white;

	cursor: pointer;
}

.button:hover {
	filter: contrast(85%);
}

.button:focus {
	outline: none;
	border: none;
}
