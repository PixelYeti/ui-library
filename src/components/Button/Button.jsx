import React from 'react';
import PropTypes from 'prop-types';

import style from './Button.css';

function Button(props) {
  return (
    <button type="button" className={style.button} style={{ backgroundColor: props.colour }} {...props}>{props.children}</button>
  );
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  colour: PropTypes.string.isRequired,
};

export default Button;
