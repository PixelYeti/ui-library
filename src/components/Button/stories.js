import React from 'react';
// Import the storybook libraries
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
// Import our component from this folder
import Button from './Button';

storiesOf('Button')
  .add('with text', () => (
    <Button onClick={action('clicked')} colour="#42ce4f">Hello</Button>
  ));
