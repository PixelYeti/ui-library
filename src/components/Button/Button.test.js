import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('Object created', () => {
    const component = renderer.create(
        <Button onClick={() => console.log(test)} colour="#42ce4f">Hello</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})