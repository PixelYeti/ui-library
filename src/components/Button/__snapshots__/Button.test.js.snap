// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object created 1`] = `
<button
  className="button"
  colour="#42ce4f"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#42ce4f",
    }
  }
  type="button"
>
  Hello
</button>
`;
